### PR TITLE
updates composer to use the fixed version of core with the `setIntent…

### PR DESCRIPTION
**Pull request by stu-greenshoots**
…` method correctly placed in the OpenDialogMessage interface.

Also updates the test that should have caught this in the first place

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -30,7 +29,7 @@ jobs:
           # fallback to using the latest cache if no exact match is found
           - v1-dependencies-
 
-      - run: composer install -n --prefer-dist --no-scripts
+      - run: composer install -n --prefer-dist --no-scripts --ignore-platform-reqs
```
Comment (MarioFrisuelos) [inline .circleci/config.yml:32]: We always should skip the use of: --ignore-platform-reqs option but if is temporally fix is ok

File: src/ResponseEngine/Message/Xmpp/XmppMessage.php [hunk 2]
```diff
@@ -124,4 +126,13 @@ public function setInternal()
     {
         return;
     }
+
+    /**
+     * @inheritDoc
```
Comment (MarioFrisuelos) [inline src/ResponseEngine/Message/Xmpp/XmppMessage.php:131]: We can improve the function adding the @var definition in the comment area.

**Review by MarioFrisuelos — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (stu-greenshoots) [reply .circleci/config.yml:32]: Yes, for now, this is a temp fix as composer on the build server bumped up to version 2 (the install script was running composer self-update) and it was complaining that grumphp does not work with composer 2

File: src/ResponseEngine/Message/Xmpp/XmppMessage.php (hunk 2 above)
Comment (stu-greenshoots) [reply src/ResponseEngine/Message/Xmpp/XmppMessage.php:131]: I've used `@ inheritDoc` here which I like to use to refer to the parent doc when nothing needs to be changed. Happy to discuss alternatives 👍

**Pull request merged**